broadinstitute/single_cell_portal_core
# Removing TCell from application (SCP-5638)

#### BACKGROUND & CHANGES
This update removes `tcell_agent` from the application stack.  This service is being replaced by Google Cloud Armor policies, which are being implemented by AppSec staff.

#### MANUAL TESTING
Since TCell is configured and loaded only in deployed environments, the only manual testing we can do is validate that the library has been removed from the stack locally.  To do this:

1. Pull the most recent copy of the `development` branch
2. Ensure `tcell_agent` is part of the Rails bundle by running this in the terminal (_not_ the Rails console): 
````
bundle show tcell_agent
=> /Users/bistline/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/tcell_agent-2.7.0
````
3. Pull this branch (`jb-tell-removal`)
4. Show that `tcell_agent` is no longer in the bundle:
```
bundle show tcell_agent
=> Could not find gem 'tcell_agent'.
```
5. (Optional) Run a global search in your IDE for `tcell` and confirm the only hits you see are for tests with things like `human_tcell_response`.
